test/test_matmul_scheduler.cpp

```diff
@@ -15,16 +15,971 @@
+  if (DataType::BFloat16 == in_type || DataType::BFloat16 == out_type) {
+    NVFUSER_TEST_CUDA_ARCH_GUARD(8, 0);
+  }
+
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  // A - tv0, B - tv1, C - tv2
+  auto tv0 = makeContigTensor(2, in_type);
+  auto tv1 = makeContigTensor(2, in_type);
+  auto tv2 = makeContigTensor(1, out_type);
+
+  // tv3 := A x B
+  auto tv3 = matmul(tv0, tv1, layout, true);
+  auto tv4 = default_out_type ? tv3 : castOp(out_type, tv3);
+
+  // tv5 := (A x B) + bias
+  auto tv5 = biasEpilogue(tv4, tv2);
+
+  // tv6 := gelu((A x B) + bias)
+  auto tv6 = gelu(tv5);
+  auto tv7 = default_out_type ? tv6 : castOp(DataType::Half, tv6);
+
+  fusion->addInput(tv0);
+  fusion->addInput(tv1);
+  fusion->addInput(tv2);
+  fusion->addOutput(tv5);
+  fusion->addOutput(tv7);
+
+  NVF_CHECK(
+      1 == ir_utils::getOpsOfType<MmaOp>(fusion.get()).size(),
+      "matmul fusion must have at least one MmaOp");
+  NVF_CHECK(
+      ir_utils::getOpsOfType<MmaOp>(fusion.get()).front()->layout().has_value(),
+      "input layout has not be set for MmaOp");
+  NVF_CHECK(
+      MatmulLayout::TN ==
+          ir_utils::getOpsOfType<MmaOp>(fusion.get()).front()->layout().value(),
+      "the MmaOp layout of Ampere MMA must always be TN");
+
+  const auto fusion_layout = mma_utils::getMatmulLayout(fusion.get());
+  NVF_CHECK(
+      fusion_layout.isValid(),
+      "failed to get decide matmul layout through fusion definition");
+  NVF_CHECK(
+      fusion_layout.getData() == layout,
+      "mismatch between test layout (",
+      toString(layout),
+      ") and layout inferred from fusion definition (",
+      toString(fusion_layout.getData()),
+      ")");
+
+  FusionExecutorCache executor_cache(std::move(fusion));
+
+  const int M = 504, N = 136, K = 248;
+
+  at::manual_seed(0);
+  auto t0 = matmulAtInput(layout, TensorMatmulPos::A, at_in_type, M, N, K);
+  auto t1 = matmulAtInput(layout, TensorMatmulPos::B, at_in_type, M, N, K);
+  auto t2 = matmulAtInput(layout, TensorMatmulPos::Bias, at_out_type, M, N, K);
+
+  auto t3 = atMatmul(t0.to(at::kFloat), t1.to(at::kFloat), layout);
+  auto t4 = default_out_type ? t3 : t3.to(at_out_type);
+  auto t5 = atBiasEpilogue(t4, t2);
+  auto t7 = at::gelu(t5);
+
+  auto outputs = executor_cache.runFusionWithInputs({t0, t1, t2});
+
+  NVF_CHECK(
+      !executor_cache.getMostRecentKernelRuntime()->isSegmented(),
+      "segmentation did happen");
+
+  // NOTE: increasted absolute tolerance to silence false negative verification
+  //       caused by different way of calculating reference
+  // D tensor results
+  NVF_CHECK(outputs[0].allclose(t5, abs_err_thr, rel_err_thr));
+  // Aux tensor results
+  NVF_CHECK(outputs[1].allclose(t7, abs_err_thr, rel_err_thr));
+}
+
 } // namespace
 
-// Matmul test that relies on segmenter for 'C = A x B' fusion,
+INSTANTIATE_TEST_CASE_P(
+    MatmulSchedulerTest,
+    PrecisionParametrizedTest,
+    ::testing::Values(HSS, HSH, TSS, TST));
```

Comment:
Should we give the tests a better name, instead of calling them things like `EpilogueBias/0`
See:
https://github.com/NVIDIA/Fuser/blob/52b98c77cfe312ab7a60171f4befd292c42c234f/test/test_memory.cpp#L103-L114

Reply:
Thanks for the hint, I'm new to this part of gtest suite, but you are right, referencing tests by numbers rather than types is not necessarily a good thing. Will be fixed in the upcoming changes.

Thanks!